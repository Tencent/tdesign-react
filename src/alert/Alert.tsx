import React, { forwardRef, useRef } from 'react';
import classNames from 'classnames';
import {
  CloseIcon as TdCloseIcon,
  InfoCircleFilledIcon as TdInfoCircleFilledIcon,
  CheckCircleFilledIcon as TdCheckCircleFilledIcon,
  ErrorCircleFilledIcon as TdErrorCircleFilledIcon,
} from 'tdesign-icons-react';
import { CSSTransition } from 'react-transition-group';
import noop from '../_util/noop';
import parseTNode from '../_util/parseTNode';
import useConfig from '../hooks/useConfig';
import useGlobalIcon from '../hooks/useGlobalIcon';
import { useLocaleReceiver } from '../locale/LocalReceiver';
import { TdAlertProps } from './type';
import { StyledProps } from '../common';
import { alertDefaultProps } from './defaultProps';
import composeRefs from '../_util/composeRefs';

const transitionTime = 200;

export interface AlertProps extends TdAlertProps, StyledProps {}

const Alert = forwardRef((props: AlertProps, ref: React.Ref<HTMLDivElement>) => {
  const { classPrefix } = useConfig();
  const [local, t] = useLocaleReceiver('alert');
  const { CloseIcon, InfoCircleFilledIcon, CheckCircleFilledIcon, ErrorCircleFilledIcon } = useGlobalIcon({
    CloseIcon: TdCloseIcon,
    InfoCircleFilledIcon: TdInfoCircleFilledIcon,
    CheckCircleFilledIcon: TdCheckCircleFilledIcon,
    ErrorCircleFilledIcon: TdErrorCircleFilledIcon,
  });

  const {
    message,
    title,
    operation,
    theme,
    icon,
    close,
    maxLine,
    onClose,
    className,
    onClosed = noop,
    ...alertProps
  } = props;

  const [closed, setClosed] = React.useState(false);
  const [collapsed, setCollapsed] = React.useState(false);

  const iconMap = {
    success: CheckCircleFilledIcon,
    info: InfoCircleFilledIcon,
    error: ErrorCircleFilledIcon,
    warning: ErrorCircleFilledIcon,
  };

  const handleClose = (e: React.MouseEvent<HTMLDivElement>) => {
    setClosed(true);
    onClose?.({ e });
  };

  const handleCollapse = () => {
    setCollapsed(!collapsed);
  };

  const renderIconNode = () => {
    if (React.isValidElement(icon)) return icon;
    return React.createElement(iconMap[theme] || iconMap.info);
  };

  const renderMessage = () => {
    if (+maxLine > 0 && Array.isArray(message)) {
      return (
        <div className={`${classPrefix}-alert__description`}>
          {message.map((item, index) => {
            if (!collapsed) {
              if (index < maxLine) {
                return <div key={index}>{item}</div>;
              }
            } else {
              return <div key={index}>{item}</div>;
            }
            return true;
          })}
          {+maxLine > 0 ? (
            <div className={`${classPrefix}-alert__collapse`} onClick={handleCollapse}>
              {!collapsed ? t(local.expandText) : t(local.collapseText)}
            </div>
          ) : null}
        </div>
      );
    }
    return <div className={`${classPrefix}-alert__description`}>{message}</div>;
  };

  const renderClose = () => (
    <div className={`${classPrefix}-alert__close`} onClick={handleClose}>
      {typeof close === 'boolean' ? <CloseIcon /> : parseTNode(close)}
    </div>
  );

  const nodeRef = useRef<HTMLDivElement>();

  return (
    <CSSTransition
      in={!closed}
      unmountOnExit
      classNames={{
        exitActive: `${classPrefix}-alert--closing`,
      }}
      nodeRef={nodeRef}
      timeout={transitionTime}
      onExited={onClosed}
    >
      <div
        ref={composeRefs(ref, nodeRef)}
        className={classNames(`${classPrefix}-alert`, `${classPrefix}-alert--${theme}`, className)}
        {...alertProps}
      >
        <div className={`${classPrefix}-alert__icon`}>{renderIconNode()}</div>
        <div className={`${classPrefix}-alert__content`}>
          {title ? <div className={`${classPrefix}-alert__title`}>{title}</div> : null}
          <div className={`${classPrefix}-alert__message`}>
            {renderMessage()}
            {operation ? <div className={`${classPrefix}-alert__operation`}>{parseTNode(operation)}</div> : null}
          </div>
        </div>
        {close ? renderClose() : null}
      </div>
    </CSSTransition>
  );
});

Alert.displayName = 'Alert';
Alert.defaultProps = alertDefaultProps;

export default Alert;
