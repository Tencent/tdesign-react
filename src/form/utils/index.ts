import type { NamePath } from '../type';

// 获取 formMap 管理的数据
export function getMapValue(name: NamePath, formMapRef: React.MutableRefObject<Map<any, any>>) {
  if (!formMapRef.current) return;

  // 提取所有 map key
  const mapKeys = [...formMapRef.current.keys()];
  // 转译为字符串后比对 key 兼容数组格式
  const key = mapKeys.find((key) => String(key) === String(name));
  // 拿到 key 引用地址获取 value
  return formMapRef.current.get(key);
}

// 将数据整理成对象，数组 name 转化嵌套对象: ['user', 'name'] => { user: { name: '' } }
export function calcFieldValue(name: NamePath, value: any) {
  if (Array.isArray(name)) {
    const fieldValue = name.reduceRight((prev, curr) => {
      const arr = [];
      if (/^\d+$/.test(String(curr))) arr[curr] = prev;
      return arr.length ? arr : { [curr]: prev };
    }, value);
    return { ...fieldValue };
  }

  return { [name]: value };
}

// 通过对象数据类型获取 map 引用: { user: { name: '' } } => formMap.get(['user', 'name'])
export function travelMapFromObject(
  obj: Record<any, any>,
  formMapRef: React.MutableRefObject<Map<any, any>>,
  callback: Function,
) {
  for (const [mapName, formItemRef] of formMapRef.current.entries()) {
    // 支持嵌套数据结构
    if (Array.isArray(mapName)) {
      // 创建唯一临时变量 symbol
      const symbol = Symbol('name');
      let fieldValue = null;

      for (let i = 0; i < mapName.length; i++) {
        const item = mapName[i];
        if (Reflect.has(fieldValue || obj, item)) {
          fieldValue = Reflect.get(fieldValue || obj, item);
        } else {
          // 当反射无法获取到值则重置为 symbol
          fieldValue = symbol;
          break;
        }
      }

      // 非 symbol 说明获取到了值
      if (fieldValue !== symbol) {
        callback(formItemRef, fieldValue);
      }
    } else if (Reflect.has(obj, mapName)) {
      callback(formItemRef, obj[mapName]);
    }
  }
}
