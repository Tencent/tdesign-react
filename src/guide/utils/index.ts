import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
