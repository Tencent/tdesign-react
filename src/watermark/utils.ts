const toLowercaseSeparator = (key: string) => key.replace(/([A-Z])/g, '-$1').toLowerCase();

// style对象转字符串
export const getStyleStr = (style: React.CSSProperties): string =>
  Object.keys(style)
    .map((key: keyof React.CSSProperties) => `${toLowercaseSeparator(key)}: ${style[key]};`)
    .join(' ');
