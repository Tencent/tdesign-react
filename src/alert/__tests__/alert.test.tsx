import React from 'react';
import { render, fireEvent, waitFor, mockTimeout, act, vi } from '@test/utils';
import Alert from '../Alert';

describe('Alert 组件测试', () => {
  const testId = 'alert-test-id';
  const text = 'Alert内容';

  test('Alert 关闭操作', async () => {
    const onClose = vi.fn();
    const onClosed = vi.fn();

    const { queryByTestId, container } = render(
      <Alert
        theme="error"
        message={text}
        close={<div data-testid={testId}>{text}</div>}
        onClose={onClose}
        onClosed={onClosed}
      />,
    );
    expect(container.querySelector('.t-alert--closing')).not.toBeInTheDocument();

    act(() => {
      fireEvent.click(queryByTestId(testId));
    });
    expect(onClose).toHaveBeenCalledTimes(1);
    expect(container.querySelector('.t-alert--closing')).toBeInTheDocument();

    await mockTimeout(() => expect(onClosed).toHaveBeenCalledTimes(1), 300);
  });

  test('Alert 展开收起操作', async () => {
    const massage = [
      <div key={0}>{text}</div>,
      <div key={1}>{text}</div>,
      <div key={2} data-testid={testId}>
        {text}
      </div>,
    ];
    const { queryByText, queryByTestId } = render(<Alert theme="error" message={massage} maxLine={2} />);
    const element = await waitFor(() => queryByTestId(testId));
    expect(element).toBeNull();

    const btn = await waitFor(() => queryByText('展开更多'));
    act(() => {
      fireEvent.click(btn);
    });

    const element1 = await waitFor(() => queryByTestId(testId));
    expect(element1).not.toBeNull();

    const btn1 = await waitFor(() => queryByText('收起'));
    act(() => {
      fireEvent.click(btn1);
    });

    const element3 = await waitFor(() => queryByTestId(testId));
    expect(element3).toBeNull();
  });
});
