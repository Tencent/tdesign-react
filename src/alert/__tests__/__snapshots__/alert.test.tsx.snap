// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert 组件测试 Alert 主题图标显示 1`] = `
<DocumentFragment>
  <div
    class="t-alert t-alert--error"
  >
    <div
      class="t-alert__icon"
    >
      <svg
        class="t-icon t-icon-error-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-alert__content"
    >
      <div
        class="t-alert__message"
      >
        <div
          class="t-alert__description"
        >
          Alert内容
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert 组件测试 Alert 主题图标显示 2`] = `
<DocumentFragment>
  <div
    class="t-alert t-alert--warning"
  >
    <div
      class="t-alert__icon"
    >
      <svg
        class="t-icon t-icon-error-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-alert__content"
    >
      <div
        class="t-alert__message"
      >
        <div
          class="t-alert__description"
        >
          Alert内容
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert 组件测试 Alert 主题图标显示 3`] = `
<DocumentFragment>
  <div
    class="t-alert t-alert--info"
  >
    <div
      class="t-alert__icon"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-alert__content"
    >
      <div
        class="t-alert__message"
      >
        <div
          class="t-alert__description"
        >
          Alert内容
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert 组件测试 Alert 主题图标显示 4`] = `
<DocumentFragment>
  <div
    class="t-alert t-alert--success"
  >
    <div
      class="t-alert__icon"
    >
      <svg
        class="t-icon t-icon-check-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 15A7 7 0 108 1a7 7 0 000 14zM4.5 8.2l.7-.7L7 9.3l3.8-3.8.7.7L7 10.7 4.5 8.2z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-alert__content"
    >
      <div
        class="t-alert__message"
      >
        <div
          class="t-alert__description"
        >
          Alert内容
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert 组件测试 Alert 主题图标显示 5`] = `
<DocumentFragment>
  <div
    class="t-alert t-alert--error"
  >
    <div
      class="t-alert__icon"
    >
      <svg
        class="t-icon t-icon-error-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-alert__content"
    >
      <div
        class="t-alert__message"
      >
        <div
          class="t-alert__description"
        >
          Alert内容
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert 组件测试 Alert 主题图标显示 6`] = `
<DocumentFragment>
  <div
    class="t-alert t-alert--error"
  >
    <div
      class="t-alert__icon"
    >
      <span
        class=""
      >
        CustomIcon
      </span>
    </div>
    <div
      class="t-alert__content"
    >
      <div
        class="t-alert__message"
      >
        <div
          class="t-alert__description"
        >
          Alert内容
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="t-alert t-alert--success"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-check-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM4.5 8.2l.7-.7L7 9.3l3.8-3.8.7.7L7 10.7 4.5 8.2z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条成功的消息提示
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--info"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-info-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条普通的消息提示
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--warning"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条警示消息
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--error"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            高危操作/出错信息提示
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`close.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="t-alert t-alert--success"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-check-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM4.5 8.2l.7-.7L7 9.3l3.8-3.8.7.7L7 10.7 4.5 8.2z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条成功的消息提示
          </div>
        </div>
      </div>
      <div
        class="t-alert__close"
      >
        <svg
          class="t-icon t-icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
    </div>
    <div
      class="t-alert t-alert--info"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-info-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条普通的消息提示
          </div>
        </div>
      </div>
      <div
        class="t-alert__close"
      >
        关闭
      </div>
    </div>
    <div
      class="t-alert t-alert--warning"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条警示消息
          </div>
        </div>
      </div>
      <div
        class="t-alert__close"
      >
        知道了
      </div>
    </div>
    <div
      class="t-alert t-alert--error"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            高危操作/出错信息提示
          </div>
        </div>
      </div>
      <div
        class="t-alert__close"
      >
        <span>
          自定义关闭
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`collapse.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-alert t-alert--info"
  >
    <div
      class="t-alert__icon"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-alert__content"
    >
      <div
        class="t-alert__message"
      >
        <div
          class="t-alert__description"
        >
          <div>
            1.这是一条普通的消息提示描述，
          </div>
          <div>
            2.这是一条普通的消息提示描述，
          </div>
          <div
            class="t-alert__collapse"
          >
            展开更多
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="t-alert t-alert--success"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-check-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM4.5 8.2l.7-.7L7 9.3l3.8-3.8.7.7L7 10.7 4.5 8.2z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条成功的消息提示
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--info"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-info-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条普通的消息提示
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--warning"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条警示消息
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--error"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            高危操作/出错信息提示
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`operation.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="t-alert t-alert--success"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-check-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM4.5 8.2l.7-.7L7 9.3l3.8-3.8.7.7L7 10.7 4.5 8.2z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条成功的消息提示
          </div>
          <div
            class="t-alert__operation"
          >
            <span>
              相关操作
            </span>
          </div>
        </div>
      </div>
      <div
        class="t-alert__close"
      >
        <svg
          class="t-icon t-icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
    </div>
    <div
      class="t-alert t-alert--info"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-info-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条普通的消息提示
          </div>
          <div
            class="t-alert__operation"
          >
            <span>
              相关操作
            </span>
          </div>
        </div>
      </div>
      <div
        class="t-alert__close"
      >
        <svg
          class="t-icon t-icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
    </div>
    <div
      class="t-alert t-alert--warning"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条警示消息
          </div>
          <div
            class="t-alert__operation"
          >
            <span>
              相关操作
            </span>
          </div>
        </div>
      </div>
      <div
        class="t-alert__close"
      >
        <svg
          class="t-icon t-icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
    </div>
    <div
      class="t-alert t-alert--error"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            高危操作/出错信息提示
          </div>
          <div
            class="t-alert__operation"
          >
            <span>
              相关操作
            </span>
          </div>
        </div>
      </div>
      <div
        class="t-alert__close"
      >
        <svg
          class="t-icon t-icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`swiper.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="t-alert t-alert--success"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-check-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM4.5 8.2l.7-.7L7 9.3l3.8-3.8.7.7L7 10.7 4.5 8.2z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条成功的消息提示
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--info"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-info-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条普通的消息提示
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--warning"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            这是一条警示消息
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-alert t-alert--error"
    >
      <div
        class="t-alert__icon"
      >
        <svg
          class="t-icon t-icon-error-circle-filled"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-alert__content"
      >
        <div
          class="t-alert__message"
        >
          <div
            class="t-alert__description"
          >
            高危操作/出错信息提示
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`title.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-alert t-alert--info"
  >
    <div
      class="t-alert__icon"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-alert__content"
    >
      <div
        class="t-alert__title"
      >
        这是一条普通的消息提示
      </div>
      <div
        class="t-alert__message"
      >
        <div
          class="t-alert__description"
        >
          这是一条普通的消息提示描述，这是一条普通的消息提示描述
        </div>
        <div
          class="t-alert__operation"
        >
          <span>
            相关操作
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
